crates/runtime/src/lib.rs

```diff
@@ -52,5 +52,12 @@ pub use crate::vmcontext::{
     VMTableDefinition, VMTableImport,
 };
 
+#[cfg(not(feature = "profiling"))]
+/// Support for conditional profiling build.
+#[derive(Clone)]
+pub struct JitDumpAgent {
```

Comment:
It is hard to read what `JitDumpAgent` is. Can we define a trait e.g. `ProfilerInfoAgent` and add `NullProfilerInfoAgent`? I can see that trait needs to have only `module_load()`.

`#[cfg(feature = "profiling")]` can introduce `JitDumpAgent` here and implement the above trait.